pandas/core/indexes/multi.py

```diff
@@ -3544,8 +3544,13 @@ def equals(self, other: object) -> bool:
             if len(self_values) == 0 and len(other_values) == 0:
                 continue
 
-            if not array_equivalent(self_values, other_values):
-                return False
+            if not isinstance(self_values, np.ndarray):
```

Comment:
maybe should do this check *in* `array_equivalent`?

Reply:
at some point probably; need to check if other usages will be affected by the semantics of EA.equals